pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -61,18 +61,28 @@ func launchTemplateName(clusterName string, arch string) string {
 	return fmt.Sprintf(launchTemplateNameFormat, clusterName, arch)
 }
 
-func (p *LaunchTemplateProvider) Get(ctx context.Context, cluster *v1alpha1.ClusterSpec, constraints Constraints) (*ec2.LaunchTemplate, error) {
+func (p *LaunchTemplateProvider) Get(ctx context.Context, cluster *v1alpha1.ClusterSpec, constraints Constraints) (*LaunchTemplateDescriptor, error) {
+	// If the customer specified a launch template then just use it
+	if desc := constraints.GetLaunchTemplate(); desc != nil {
+		return desc, nil
+	}
+
 	arch := utils.NormalizeArchitecture(constraints.Architecture)
 	name := launchTemplateName(cluster.Name, *arch)
+	version := "$Default"
+	result := &LaunchTemplateDescriptor{
+		Version: &version,
+	}
 	if launchTemplate, ok := p.cache.Get(name); ok {
```

Comment:
Should we be caching Id+Version as the key?

Reply:
IIUC version is always `$Default` when querying using the EC2 API.  If a version were specified by the user, it would short circuit at the top of the func.  

Do you think there'd be a future feature where customers could specify to use `$Latest` instead of default by just providing the the launch template version and no launch template name?

Reply:
I was thinking we would not support that - seems like too much magic and assumptions about what we're doing with the launch templates ourselves. WDYT?

Reply:
@ellistarn I'd prefer to leave it for now as it is always $Default, could add that in later if we ever change.

Reply:
> I was thinking we would not support that - seems like too much magic and assumptions about what we're doing with the launch templates ourselves. WDYT?

👍  Supporting just default seems fine to me, was just thinking on why we'd need need to cache with the version key.